From 52dc740b2191ac29f7149fa1eda7d7d1ba1e4ef2 Mon Sep 17 00:00:00 2001
From: Wang Haoran <b50027@freescale.com>
Date: Tue, 11 Nov 2014 10:02:48 +0800
Subject: [PATCH 08/22] MA-5839-2 Fix compile error for upgrade Android-L

Choose the correct libnl for wpa_supplicant private
lib.

Signed-off-by: Wang Haoran <b50027@freescale.com>

Conflicts:
	hostapd/Android.mk
	wpa_supplicant/Android.mk
---
 hostapd/Android.mk        | 24 ++++++++++++++++++++++++
 wpa_supplicant/Android.mk | 26 ++++++++++++++++++++++++++
 2 files changed, 50 insertions(+)

diff --git a/hostapd/Android.mk b/hostapd/Android.mk
index 3232630..a4fa0b6 100644
--- a/hostapd/Android.mk
+++ b/hostapd/Android.mk
@@ -935,4 +935,28 @@ LOCAL_SRC_FILES := $(OBJS)
 LOCAL_C_INCLUDES := $(INCLUDES)
 include $(BUILD_EXECUTABLE)
 
+########################
+include $(CLEAR_VARS)
+LOCAL_MODULE := rtl_hostapd
+LOCAL_MODULE_TAGS := optional
+ifdef CONFIG_DRIVER_CUSTOM
+LOCAL_STATIC_LIBRARIES := libCustomWifi
+endif
+ifneq ($(BOARD_HOSTAPD_PRIVATE_LIB_RTL),)
+LOCAL_STATIC_LIBRARIES += $(BOARD_HOSTAPD_PRIVATE_LIB_RTL)
+endif
+LOCAL_SHARED_LIBRARIES := libc libcutils liblog libcrypto libssl
+ifdef CONFIG_DRIVER_NL80211
+ifneq ($(wildcard external/libnl),)
+LOCAL_SHARED_LIBRARIES += libnl
+else
+LOCAL_STATIC_LIBRARIES += libnl_2
+endif
+endif
+LOCAL_CFLAGS := $(L_CFLAGS)
+LOCAL_CFLAGS += -DREALTEK_WIFI_VENDOR
+LOCAL_SRC_FILES := $(OBJS)
+LOCAL_C_INCLUDES := $(INCLUDES)
+include $(BUILD_EXECUTABLE)
+
 endif # ifeq ($(WPA_BUILD_HOSTAPD),true)
diff --git a/wpa_supplicant/Android.mk b/wpa_supplicant/Android.mk
index 9356b6c..bb24c18 100644
--- a/wpa_supplicant/Android.mk
+++ b/wpa_supplicant/Android.mk
@@ -1582,6 +1582,32 @@ LOCAL_C_INCLUDES := $(INCLUDES)
 include $(BUILD_EXECUTABLE)
 
 ########################
+include $(CLEAR_VARS)
+LOCAL_MODULE := rtl_wpa_supplicant
+ifdef CONFIG_DRIVER_CUSTOM
+LOCAL_STATIC_LIBRARIES := libCustomWifi
+endif
+ifneq ($(BOARD_WPA_SUPPLICANT_PRIVATE_LIB_RTL),)
+LOCAL_STATIC_LIBRARIES += $(BOARD_WPA_SUPPLICANT_PRIVATE_LIB_RTL)
+endif
+LOCAL_SHARED_LIBRARIES := libc libcutils liblog
+ifeq ($(CONFIG_TLS), openssl)
+LOCAL_SHARED_LIBRARIES += libcrypto libssl libkeystore_binder
+endif
+ifdef CONFIG_DRIVER_NL80211
+ifneq ($(wildcard external/libnl),)
+LOCAL_SHARED_LIBRARIES += libnl
+else
+LOCAL_STATIC_LIBRARIES += libnl_2
+endif
+endif
+LOCAL_CFLAGS := $(L_CFLAGS)
+LOCAL_CFLAGS += -DREALTEK_WIFI_VENDOR
+LOCAL_SRC_FILES := $(OBJS)
+LOCAL_C_INCLUDES := $(INCLUDES)
+include $(BUILD_EXECUTABLE)
+
+########################
 #
 #include $(CLEAR_VARS)
 #LOCAL_MODULE := eapol_test
-- 
1.8.0

